### PR TITLE
id, expiry and rewardsClaimed

### PR DESCRIPTION
**Pull request by robrobbins**
#73 #72

### CODE REVIEW

File: contracts/Parameterizer.sol [hunk 1]
```diff
@@ -168,47 +168,48 @@ contract Parameterizer {
...
     ParamProposal memory prop = proposals[_propID];
     uint deposit = prop.deposit;
 
-    require(propExists(_propID) && prop.challengeID == 0);
+    require(propExists(_propID) && prop.challenge == 0);
 
-    //start poll
-    uint pollID = voting.startPoll(
+    //start poll, uint declared as return
+    id = voting.startPoll(
```
Comment (geoffkinder) [inline contracts/Parameterizer.sol:180]: Maybe I’m missing it, but is `id` declared anywhere? Does the return value definition count as a variable declaration in solidity?

**Review by geoffkinder — CHANGES_REQUESTED** [review]
No blockers with the changes in this PR, but I don’t think it fully addresses the items listed in https://github.com/computablelabs/computable/issues/73. I went ahead and checked the boxes for the items that have been addressed thus far. I also realize that the `Computable.js` updates will be a separate PR.

**Comment by geoffkinder** [comment]
I think the struct updates don’t matter as much, or possibly don’t make sense after our discussion about `Challenge` and `Poll` yesterday.

The removal of `_data` from `Registry.challenge()` is just clean up work that can be deferred until later. It is just an extra param that appears to not be used at all.

I think the only remaining item that I would like in this PR is the addition of `_listingHash` to the `Registry` `Challenge` event, so I can trace the challenge back to the listing it is associated with. We currently only have data pointing the other direction, from the listing to the challenge.

**Comment by rbharath** [comment]
Did a drive-by quick skim and I like the changes. Simplifies some of the naming structure and makes it easier to follow. Deferring on actual code review to @SUICIDALcow :)

File: contracts/Parameterizer.sol (hunk 1 above)
Comment (robrobbins) [reply contracts/Parameterizer.sol:180]: yes. as a matter of fact if you tried to declare it here - it would error

**Comment by robrobbins** [comment]
the listinghash is published with the registry challenge event

**Comment by robrobbins** [comment]
and if you ever needed the *Expirys related to a challenge you could call `voting.pollMap(id)` - they are in there

**Review by geoffkinder — APPROVED** [review]

**Pull request merged**